flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -346,7 +346,11 @@ object GenerateUtils {
         ctx.addReusableMember(fieldDecimal)
         val value = Decimal.fromBigDecimal(
           literalValue.asInstanceOf[JBigDecimal], precision, scale)
-        generateNonNullLiteral(literalType, fieldTerm, value)
+        if (value == null) {
```

Comment:
Just `generateLiteral`?

Reply:
That won't work because we expect a decimal literal to be a `BigDecimal`. But now the value has been casted to a `Decimal `.

Reply:
So we should improve `generateLiteral`, and I think maybe this is risk in it too.